engines/fact_check/spec/request/blocks_controller_spec.rb

```diff
@@ -0,0 +1,83 @@
+RSpec.describe FactCheck::BlocksController, type: :request do
+  include FactCheck::Engine.routes.url_helpers
+
+  describe "#show" do
+    let(:block) { build(:content_block) }
+    let(:content_id) { "some-content-block" }
+    let(:user) { create(:user) }
+
+    before do
+      ENV["JWT_AUTH_SECRET"] = "secret"
+      allow(ContentBlock).to receive(:from_content_id_alias).with(content_id).and_return(block)
+    end
+
+    context "when the user is authenticated" do
+      before do
+        logout
+        login_as(user)
+      end
+
+      it "returns success" do
+        get block_path(content_id)
+
+        expect(response).to have_http_status(:success)
+        expect(response).to render_template(:show)
+      end
+    end
+
+    context "when the user is not authenticated" do
+      before do
+        logout
+      end
+
+      around do |test|
+        ClimateControl.modify("GDS_SSO_MOCK_INVALID" => "1") do
```

Comment:
is it worth including a note or comment here to explain this `GDS_SSO_MOCK_INVALID` environment variable, as I for one wasn't clear what this does? Looking around (and from  https://github.com/alphagov/gds-sso/blob/f1175386c48ae4e79bf843ef9660aba7abf9e4e1/lib/gds-sso/warden_config.rb#L102) I see that it short-circuits the "mock" Warden SSO strategy so that the Signon auth fails.